sound/soc/sof/intel/hda-dsp.c

```diff
@@ -282,6 +282,52 @@ void hda_dsp_ipc_int_disable(struct snd_sof_dev *sdev)
 			HDA_DSP_REG_HIPCCTL_BUSY | HDA_DSP_REG_HIPCCTL_DONE, 0);
 }
 
+static void hda_dsp_wait_d0i3c_done(struct snd_sof_dev *sdev, int *retry)
```

Comment:
this handles retries so should return an error if there is no change after all the retries